# The confidence level is 0.05.

In [2]:
from __future__ import print_function
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import pylab as py
import os
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [213]:
import scipy.stats
import sys

In [113]:
url = "/gws/open/Student/citibike/201707-citibike-tripdata.csv.zip"
df = pd.read_csv(url)
df.head()

tripduration            starttime             stoptime  start station id  \
0           364  2017-07-01 00:00:00  2017-07-01 00:06:05               539   
1          2142  2017-07-01 00:00:03  2017-07-01 00:35:46               293   
2           328  2017-07-01 00:00:08  2017-07-01 00:05:37              3242   
3          2530  2017-07-01 00:00:11  2017-07-01 00:42:22              2002   
4          2534  2017-07-01 00:00:15  2017-07-01 00:42:29              2002   

               start station name  start station latitude  \
0  Metropolitan Ave & Bedford Ave               40.715348   
1           Lafayette St & E 8 St               40.730207   
2      Schermerhorn St & Court St               40.691029   
3    Wythe Ave & Metropolitan Ave               40.716887   
4    Wythe Ave & Metropolitan Ave               40.716887   

   start station longitude  end station id          end station name  \
0               -73.960241            3107  Bedford Ave & Nassau Ave   
1               -73.991026            3425         2 Ave  & E 104 St   
2               -73.991834            3397      Court St & Nelson St   
3               -73.963198             398  Atlantic Ave & Furman St   
4               -73.963198             398  Atlantic Ave & Furman St   

   end station latitude  end station longitude  bikeid    usertype  \
0             40.723117             -73.952123   14744  Subscriber   
1             40.789210             -73.943708   19587  Subscriber   
2             40.676395             -73.998699   27937  Subscriber   
3             40.691652             -73.999979   26066  Subscriber   
4             40.691652             -73.999979   29408  Subscriber   

   birth year  gender  
0      1986.0       1  
1      1981.0       1  
2      1984.0       2  
3      1985.0       1  
4      1982.0       2

In [116]:
df['starttime'] = pd.to_datetime(df['starttime'])

In [117]:
df['starttime'] = df['starttime'].dt.time

In [118]:
data1=pd.DataFrame(df[['tripduration','starttime']])

In [120]:
data1['starttime'].head()

0    00:00:00
1    00:00:03
2    00:00:08
3    00:00:11
4    00:00:15
Name: starttime, dtype: object

In [121]:
day=data1[data1['starttime']<=datetime.time(20, 20, 0)]

In [122]:
day=day[day['starttime']>=datetime.time(5, 40, 0)]

In [123]:
day.head()

tripduration starttime
1564          1445  05:41:42
1565           910  05:42:24
1566           254  05:42:30
1567          1386  05:43:18
1568           179  05:43:42

In [124]:
night=data1[(data1['starttime']>datetime.time(20, 20, 0)) | (data1['starttime']<datetime.time(5, 40, 0))]

In [125]:
night.head()

tripduration starttime
0           364  00:00:00
1          2142  00:00:03
2           328  00:00:08
3          2530  00:00:11
4          2534  00:00:15

In [126]:
ks = scipy.stats.ks_2samp(day['tripduration'], night['tripduration'])
print (ks)

Ks_2sampResult(statistic=0.017460995087591691, pvalue=1.0700394410449435e-54)


# The p-value is smaller than 0.05, so the null hypothesis is rejected. Trip durations in the day and night have different distributions.

In [127]:
day1=day.sort(['tripduration']).iloc[0:70000,:]

/opt/rh/anaconda/root/envs/PUI2016_Python3/lib/python3.5/site-packages/ipykernel/__main__.py:1: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)
  if __name__ == '__main__':


In [128]:
night1=night.sort(['tripduration']).iloc[0:70000,:]

/opt/rh/anaconda/root/envs/PUI2016_Python3/lib/python3.5/site-packages/ipykernel/__main__.py:1: FutureWarning: sort(columns=....) is deprecated, use sort_values(by=.....)
  if __name__ == '__main__':


In [129]:
stats.spearmanr(day1['tripduration'], night1['tripduration'])

SpearmanrResult(correlation=0.99994401144399137, pvalue=0.0)

# P-value is smaller than 0.05, which means null hypothesis is rejected. The trip durations in the day and night have correlations.

In [131]:
stats.pearsonr(day1['tripduration'], night1['tripduration'])

(0.99563805437461472, 0.0)

# The correlation coefficient is 0.99563805437461472, and the p-values is smaller than 0.05. Thus, the null hypothesis is rejected, and there is a strong postive correlatin between the time duration in the day and night.

In [133]:
url2 = "/gws/open/Student/citibike/201706-citibike-tripdata.csv.zip"
df2 = pd.read_csv(url2)

In [134]:
jul=pd.DataFrame(df[['birth year','start station latitude','start station longitude']])
jun=pd.DataFrame(df2[['birth year','start station latitude','start station longitude']])

In [137]:
jun.head()

birth year  start station latitude  start station longitude
0      1967.0               40.760094               -73.994618
1      1981.0               40.756458               -73.993722
2      1982.0               40.735877               -73.982050
3      1973.0               40.711174               -73.996826
4      1985.0               40.707179               -74.008873

In [138]:
jul.head()

birth year  start station latitude  start station longitude
0      1986.0               40.715348               -73.960241
1      1981.0               40.730207               -73.991026
2      1984.0               40.691029               -73.991834
3      1985.0               40.716887               -73.963198
4      1982.0               40.716887               -73.963198

In [149]:
use2=pd.concat([jun,jul])
use2=use2.dropna()
use2.head()

birth year  start station latitude  start station longitude
0      1967.0               40.760094               -73.994618
1      1981.0               40.756458               -73.993722
2      1982.0               40.735877               -73.982050
3      1973.0               40.711174               -73.996826
4      1985.0               40.707179               -74.008873

In [150]:
use2['birth year']=use2['birth year'].astype(int)
use2.head()

birth year  start station latitude  start station longitude
0        1967               40.760094               -73.994618
1        1981               40.756458               -73.993722
2        1982               40.735877               -73.982050
3        1973               40.711174               -73.996826
4        1985               40.707179               -74.008873

In [167]:
use2['age']=np.add(2017,-use2['birth year'])

In [191]:
location=[]
location=use2['start station latitude'].astype(str)+ ', ' +use2['start station longitude'].astype(str)
location.head()

0    40.76009437, -73.99461843
1    40.75645824, -73.99372222
2    40.73587678, -73.98205027
3    40.71117444, -73.99682619
4    40.70717936, -74.00887308
dtype: object

In [192]:
use2['location']=location

In [193]:
use2.head()

birth year  start station latitude  start station longitude  age  \
0        1967               40.760094               -73.994618   50   
1        1981               40.756458               -73.993722   36   
2        1982               40.735877               -73.982050   35   
3        1973               40.711174               -73.996826   44   
4        1985               40.707179               -74.008873   32   

                    location  
0  40.76009437, -73.99461843  
1  40.75645824, -73.99372222  
2  40.73587678, -73.98205027  
3  40.71117444, -73.99682619  
4  40.70717936, -74.00887308

In [2]:
import shapefile

ImportError: No module named 'shapefile'

In [217]:
gpd.read_file('/wingrdp/homedirs/gc2300/PUI HW/map.shp')

NameError: name 'gpd' is not defined